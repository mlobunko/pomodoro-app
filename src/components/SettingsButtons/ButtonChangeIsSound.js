import React from "react";
import { connect } from "react-redux";
import { changeIsSound } from "../../actions/settings";

export const ButtonChangeIsSound = ({ changeIsSound, isSound }) => (
  <div onClick={changeIsSound} className="button-plus-minus">
    {isSound ? <i class="fa fa-volume-up" /> : <i class="fa fa-volume-down" />}
  </div>
);

const mapStateToProps = state => ({
  isSound: state.settings.isSound
});

export default connect(mapStateToProps, { changeIsSound })(ButtonChangeIsSound);
