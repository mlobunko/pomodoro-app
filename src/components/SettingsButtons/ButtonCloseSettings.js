import React from "react";
import { connect } from "react-redux";
import { closeSettings } from "../../actions/statistics";

export const ButtonCloseSettings = ({ closeSettings }) => (
  <button
    onClick={closeSettings}
    className="button"
    aria-label="button close settings window"
  >
    OK
  </button>
);

export default connect(undefined, { closeSettings })(ButtonCloseSettings);
