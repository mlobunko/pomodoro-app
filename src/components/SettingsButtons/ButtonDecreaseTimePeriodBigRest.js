import React from "react";
import { connect } from "react-redux";
import { decreaseTimePeriodBigRest } from "../../actions/settings";

export const ButtonDecreaseTimePeriodBigRest = ({
  decreaseTimePeriodBigRest
}) => (
  <button
    onClick={decreaseTimePeriodBigRest}
    className="button-plus-minus"
    aria-label="button decrease time for big rest"
  >
    <i className="fa fa-minus" />
  </button>
);

export default connect(undefined, { decreaseTimePeriodBigRest })(
  ButtonDecreaseTimePeriodBigRest
);
