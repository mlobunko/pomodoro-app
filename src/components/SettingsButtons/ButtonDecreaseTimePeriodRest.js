import React from "react";
import { connect } from "react-redux";
import { decreaseTimePeriodRest } from "../../actions/settings";

export const ButtonDecreaseTimePeriodRest = ({ decreaseTimePeriodRest }) => (
  <button
    onClick={decreaseTimePeriodRest}
    className="button-plus-minus"
    aria-label="button decrease time for rest"
  >
    <i className="fa fa-minus" />
  </button>
);

export default connect(undefined, { decreaseTimePeriodRest })(
  ButtonDecreaseTimePeriodRest
);
