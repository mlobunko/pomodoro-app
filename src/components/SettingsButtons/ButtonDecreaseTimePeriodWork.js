import React from "react";
import { connect } from "react-redux";
import { decreaseTimePeriodWork } from "../../actions/settings";

export const ButtonDecreaseTimePeriodWork = ({ decreaseTimePeriodWork }) => (
  <button onClick={decreaseTimePeriodWork} className="button-plus-minus">
    <i class="fa fa-minus" />
  </button>
);

export default connect(undefined, { decreaseTimePeriodWork })(
  ButtonDecreaseTimePeriodWork
);
