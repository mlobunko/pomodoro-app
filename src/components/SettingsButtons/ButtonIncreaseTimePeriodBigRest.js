import React from "react";
import { connect } from "react-redux";
import { increaseTimePeriodBigRest } from "../../actions/settings";

export const ButtonIncreaseTimePeriodBigRest = ({
  increaseTimePeriodBigRest
}) => (
  <button
    onClick={increaseTimePeriodBigRest}
    className="button-plus-minus"
    aria-label="button increase time for big rest"
  >
    <i className="fa fa-plus" />
  </button>
);

export default connect(undefined, { increaseTimePeriodBigRest })(
  ButtonIncreaseTimePeriodBigRest
);
