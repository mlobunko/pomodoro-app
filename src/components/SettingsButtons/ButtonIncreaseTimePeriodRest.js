import React from "react";
import { connect } from "react-redux";
import { increaseTimePeriodRest } from "../../actions/settings";

export const ButtonIncreaseTimePeriodRest = ({ increaseTimePeriodRest }) => (
  <button onClick={increaseTimePeriodRest} className="button-plus-minus">
    <i class="fa fa-plus" />
  </button>
);

export default connect(undefined, { increaseTimePeriodRest })(
  ButtonIncreaseTimePeriodRest
);
