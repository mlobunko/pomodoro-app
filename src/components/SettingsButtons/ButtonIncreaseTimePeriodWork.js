import React from "react";
import { connect } from "react-redux";
import { increaseTimePeriodWork } from "../../actions/settings";

export const ButtonIncreaseTimePeriodWork = ({ increaseTimePeriodWork }) => (
  <button
    onClick={increaseTimePeriodWork}
    className="button-plus-minus"
    aria-label="button increase time for work"
  >
    <i className="fa fa-plus" />
  </button>
);

export default connect(undefined, { increaseTimePeriodWork })(
  ButtonIncreaseTimePeriodWork
);
