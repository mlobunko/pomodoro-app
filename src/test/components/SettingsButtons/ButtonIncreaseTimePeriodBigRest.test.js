import React from "react";
import { shallow } from "enzyme";
import { ButtonIncreaseTimePeriodBigRest } from "../../../components/SettingsButtons/ButtonIncreaseTimePeriodBigRest";

it("should render ButtonIncreaseTimePeriodBigRest", () => {
  const wrapper = shallow(<ButtonIncreaseTimePeriodBigRest />);
  expect(wrapper).toMatchSnapshot();
});

it("should call increaseTimePeriodBigRest on a click", () => {
  const increaseTimePeriodBigRest = jest.fn();
  const wrapper = shallow(
    <ButtonIncreaseTimePeriodBigRest
      increaseTimePeriodBigRest={increaseTimePeriodBigRest}
    />
  );
  wrapper.find("button").simulate("click");
  expect(increaseTimePeriodBigRest).toHaveBeenCalledTimes(1);
});
