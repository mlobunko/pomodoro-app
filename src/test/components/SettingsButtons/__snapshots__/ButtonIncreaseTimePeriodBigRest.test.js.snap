// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ButtonIncreaseTimePeriodBigRest 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "aria-label": "button increase time for big rest",
      "children": <i
        className="fa fa-plus"
/>,
      "className": "button-plus-minus",
      "onClick": undefined,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "className": "fa fa-plus",
      },
      "ref": null,
      "rendered": null,
      "type": "i",
    },
    "type": "button",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "aria-label": "button increase time for big rest",
        "children": <i
          className="fa fa-plus"
/>,
        "className": "button-plus-minus",
        "onClick": undefined,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "fa fa-plus",
        },
        "ref": null,
        "rendered": null,
        "type": "i",
      },
      "type": "button",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
